projects/plugins/jetpack/tests/test-gutenberg-version.mjs

```diff
@@ -0,0 +1,169 @@
+	}
+
+	spawnSync(
+		'git',
+		[
+			'clone',
+			'git@github.com:WordPress/gutenberg.git',
+			'./tests/temp-gutenberg-checkout',
+			'--branch',
+			version,
+		],
+		{ stdio: 'inherit' }
+	);
+}
+
+function checkoutCommit( commit ) {
+	console.log( `Checking out commit ${ commit } from gutenberg repo` );
+	spawnSync( 'git', [ 'checkout', commit ], {
+		stdio: 'inherit',
+		cwd: './tests/temp-gutenberg-checkout',
+	} );
+}
+
+function installGutenbergDependencies() {
+	console.log( 'Installing Gutenberg dependencies' );
+	spawnSync( 'npm', [ 'install' ], { stdio: 'inherit', cwd: './tests/temp-gutenberg-checkout' } );
+}
+
+function buildGutenbergPackages() {
+	console.log( 'Building Gutenberg packages' );
+	spawnSync( 'npm', [ 'run', 'build:packages' ], {
+		stdio: 'inherit',
+		cwd: './tests/temp-gutenberg-checkout',
+	} );
+}
+
+function moveGutenbergPackages() {
+	console.log( 'Moving Gutenberg packages' );
+	return fs.move( './tests/temp-gutenberg-checkout/packages', './packages' );
+}
+
+function updatePackageJsonDependencies() {
+	console.log( 'Updating Gutenberg package.json dependencies' );
+	const gutenbergPackageJson = JSON.parse(
+		fs.readFileSync( './tests/temp-gutenberg-checkout/package.json' )
+	);
+	const jetpackPackageJson = JSON.parse( fs.readFileSync( './package.json' ) );
+
+	const wordPressDeps = Object.fromEntries(
+		Object.entries( gutenbergPackageJson.dependencies ).filter(
+			( [ key ] ) => key.includes( '@wordpress' ) || key.includes( '@emotion' )
+		)
+	);
+
+	const wordPressDevDeps = Object.fromEntries(
+		Object.entries( gutenbergPackageJson.devDependencies ).filter(
+			( [ key ] ) => key.includes( '@wordpress' ) || key.includes( '@emotion' )
+		)
+	);
+
+	jetpackPackageJson.dependencies = Object.fromEntries(
+		Object.entries( jetpackPackageJson.dependencies ).filter(
+			( [ key ] ) => ! key.includes( '@wordpress' )
+		)
+	);
+
+	jetpackPackageJson.dependencies = { ...wordPressDeps, ...jetpackPackageJson.dependencies };
+
+	jetpackPackageJson.devDependencies = Object.fromEntries(
+		Object.entries( jetpackPackageJson.devDependencies ).filter(
+			( [ key ] ) => ! key.includes( '@wordpress' )
+		)
+	);
+
+	jetpackPackageJson.devDependencies = {
+		...wordPressDevDeps,
+		...jetpackPackageJson.devDependencies,
+	};
+
+	try {
+		fs.writeFileSync( './package.json', JSON.stringify( jetpackPackageJson ) );
+	} catch ( err ) {
+		console.error( err );
+	}
+}
+
+function updateJetpackDependencies() {
+	console.log( 'Updating Jetpack dependencies' );
+	spawnSync( 'pnpm', [ 'install' ], { stdio: 'inherit' } );
+	console.log( 'Done Updating Jetpack dependencies' );
+}
+
+function installAdditionalGutenbergDependencies() {
+	console.log( 'Installing additional Gutenberg dependencies' );
+	spawnSync(
+		'pnpm',
+		[
+			'add',
+			'showdown',
+			'simple-html-tokenizer',
+			'hpq',
+			'react-autosize-textarea',
+			'traverse',
+			'css-mediaquery',
+			'@emotion/styled',
+			'@emotion/react',
```

Comment:
What's going on here? Are all of these going to have to be added someday when we get around to making #19328 happen?

Reply:
This is just temporarily adding some additional dependencies that the latest Gutenberg needs to successfully build - I assume that these could be removed when #19328 happens, although there is always the potential that the cutting edge Gutenberg is going to have some dependencies that the Jetpack repo doesn't.